lib/trie/decode/byte.go

```diff
@@ -0,0 +1,16 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package decode
+
+import "io"
+
+// ReadNextByte reads the next byte from the reader.
+func ReadNextByte(reader io.Reader) (b byte, err error) {
```

Comment:
I would've assumed this would have been some private helper function inside the `trie` package.

Reply:
See comment https://github.com/ChainSafe/gossamer/pull/2077#discussion_r760183225

Reply:
Changed in c8deadd377c4b1781e631e6e56c21066135801cc so there is no longer a helper function and it also uses sync.Pools to save those 1-byte slices allocations :wink: